### DISCUSSION TITLE
Gridlines - like PowerPoint for example

### DISCUSSION THREAD

**Discussion by raarts** [xyflow/xyflow#2091] [New Features]
This would be useful, and relatively hard to implement outside of react-flow (I think).
I mention PowerPoint because:

1. they do this pretty well (especially relative to other objects) and 
2. it's familiar to many.

**Comment by bcakmakoglu** [comment]
I am unsure what the feature request is. 
You haven't outlined what it is exactly you're trying to achieve or what you already tried and where current limitations were for you. 
I'm certain the maintainers would be thankful if you would expand a bit on that 😄

But -
You can already create a background with lines instead of dots, if that's what you meant:
`<Background variant={BackgroundVariant.Lines} />`
If you want elements to snap to a grid, that also works already with the `snapGrid` and `snapToGrid` props. 
`snapGrid` accepts an array of two numbers which represent the grid steps with which elements will be moved when dragging. 

Are these possibly the features you were looking for?

**Reply by raarts** [reply]
Sorry, I'm too used to working with PowerPoint...

In Powerpoint, while moving objects, helplines are drawn to show you if you are currently exactly in the center (horizontal or vertical) of the canvas or if the object you are dragging is exactly on the same x, or y coordinate as other objects. This allows you to line up objects exactly. It's really helpful.

Maybe best to start Powerpoint, drop some objects (circles, squares) on a slide, move the around, and observe the helplines PowerPoint shows. 

Other drawing packages do this too.

**Reply by bcakmakoglu** [reply]
Ah now I get what you mean (tbh I haven't used PowerPoint in a while 😉), it wasn't that obvious to me from your opening post, hence why I asked :)
